src/control/createIndex.ts

```diff
@@ -104,7 +103,7 @@ const CreateIndexOptionsSchema = Type.Object(
     name: IndexNameSchema,
     dimension: DimensionSchema,
     metric: MetricSchema,
-
+    deletionProtection: Type.Optional(Type.String()),
```

Comment:
Is there a way to communicate it's not any string but specifically `'enabled'` or `'disabled'` with these types? Also, out of curiosity, did you do any testing where you try passing random strings in this field? I'm curious if the API is properly validating it.

Reply:
Ah good point, I think there is with the schema stuff, let me take a look.

I did test passing random strings, and the API does seem to validate. I wrote a specific integration test for it: https://github.com/pinecone-io/pinecone-ts-client/pull/238/files#diff-2827e004959e078a377f7fd8a1662a3c3a93109869e4070fd5ad6c52acbd8257R146-R148